stdlib/arg.ml

```diff
@@ -336,3 +346,48 @@ let align ?(limit=max_int) speclist =
   let len = List.fold_left max_arg_len 0 completed in
   let len = min len limit in
   List.map (add_padding len) completed
+
+let trim_cr s =
+  let len = String.length s in
+  if len > 0 && String.get s (len - 1) = '\r' then
+    String.sub s 0 (len - 1)
+  else
+    s
+
+let read_aux trim sep file =
+  let ic = open_in_bin file in
+  let buf = Buffer.create 200 in
+  let words = ref [] in
+  let stash () =
+    let word =  (Buffer.contents buf) in
+    let word = if trim then trim_cr word else word in
+    words := word :: !words;
+    Buffer.clear buf
+  in
+  let rec read () =
+    try
+      let c = input_char ic in
+      if c = sep then begin
+        stash (); read ()
+      end else begin
+        Buffer.add_char buf c; read ()
+      end
+    with End_of_file ->
```

Comment:
FWIW, this try...with breaks tail-recursion and would lead to stack overflow if the file is large enough.  Will submit a PR to address that shortly.